internal/status/crd_status.go

```diff
@@ -51,7 +54,7 @@ func UpdateHostRuleStatus(key string, hr *akov1beta1.HostRule, updateStatus Upda
 		"status": akov1beta1.HostRuleStatus(updateStatus),
 	})
 
-	_, err := lib.AKOControlConfig().V1beta1CRDClientset().AkoV1beta1().HostRules(hr.Namespace).Patch(context.TODO(), hr.Name, types.MergePatchType, patchPayload, metav1.PatchOptions{}, "status")
+	hrFromK8sClient, err := lib.AKOControlConfig().V1beta1CRDClientset().AkoV1beta1().HostRules(hr.Namespace).Patch(context.TODO(), hr.Name, types.MergePatchType, patchPayload, metav1.PatchOptions{}, "status")
```

Comment:
initial comment: why are we retrying here as we are fetching from directly clientset so we should have updated content?

Reply:
We are doing a patch through clientset which returns the updated result but we are not sure if the crd informer is also updated with the changes. So here, we are waiting for the informer to be updated with the patch